external-crates/move/move-compiler/src/expansion/translate.rs

```diff
@@ -2572,11 +2577,24 @@ fn bind(context: &mut Context, sp!(loc, pb_): P::Bind) -> Option<E::LValue> {
         PB::Unpack(ptn, ptys_opt, pfields) => {
             let tn = name_access_chain(context, Access::ApplyNamed, *ptn)?;
             let tys_opt = optional_types(context, ptys_opt);
-            let vfields: Option<Vec<(Field, E::LValue)>> = pfields
-                .into_iter()
-                .map(|(f, pb)| Some((f, bind(context, pb)?)))
-                .collect();
-            let fields = fields(context, loc, "deconstruction binding", "binding", vfields?);
+            let fields = match pfields {
+                FieldBindings::Named(named_bindings) => {
+                    let vfields: Option<Vec<(Field, E::LValue)>> = named_bindings
+                        .into_iter()
+                        .map(|(f, pb)| Some((f, bind(context, pb)?)))
+                        .collect();
+                    let fields =
+                        named_fields(context, loc, "deconstruction binding", "binding", vfields?);
+                    E::FieldBindings::Named(fields)
+                }
+                FieldBindings::Positional(positional_bindings) => {
+                    let fields: Option<Vec<E::LValue>> = positional_bindings
```

Comment:
This is good for now, Just wondering if we should maybe replace `None`s on each inner binding with some sort of `UnresolvedError` binding